lib/utils/token_set_utils.dart

```diff
@@ -0,0 +1,53 @@
+/// Parses token sets from token data and returns a list of all available tokens
+/// sets.
+///
+/// Removed the default set from the output.
+List<String> getTokenSets(Map<String, dynamic> tokens) {
+  final tokenSets = (tokens['\$metadata']['tokenSetOrder'] as List)
+      .cast<String>()
+    ..remove('global');
+  return tokenSets;
+}
+
+/// Overrides and merges the `baseSet` with `withSet`.
+Map<String, dynamic> overrideAndMergeTokenSet(
+  Map<String, dynamic> baseSet, {
+  required Map<String, dynamic> withSet,
+}) {
+  return baseSet..addAll(withSet);
+}
+
+/// Check if `data` has any child that has type `type`.
+bool hasNestedType(MapEntry<String, dynamic> data, {required String type}) {
+  final value = data.value as Map<String, dynamic>;

```

Comment:
What if the value is not a map?

Reply:
Wouldn't it then fail when calling `values.entries` later?

Reply:
Yes true. Will be fixed with next PR. (There its already fixed)